tests/rptest/tests/enterprise_features_license_test.py

```diff
@@ -0,0 +1,213 @@
+                    "Skipping test, REDPANDA_SAMPLE_LICENSE env var not found")
+                return
+
+        if feature == Features.audit_logging:
+            self.redpanda.set_cluster_config(
+                {
+                    'audit_enabled': True,
+                },
+                expect_restart=True,
+            )
+        elif feature == Features.cloud_storage:
+            self.redpanda.set_cluster_config({'cloud_storage_enabled': 'true'},
+                                             expect_restart=True)
+        elif feature == Features.partition_auto_balancing_continuous:
+            self.redpanda.set_cluster_config(
+                {'partition_autobalancing_mode': 'continuous'})
+        elif feature == Features.core_balancing_continuous:
+            self.redpanda.set_cluster_config(
+                {'core_balancing_continuous': 'true'})
+        elif feature == Features.gssapi:
+            self.redpanda.set_cluster_config(
+                {'sasl_mechanisms': ['SCRAM', 'GSSAPI']})
+        elif feature == Features.oidc:
+            self.redpanda.set_cluster_config(
+                {'sasl_mechanisms': ['SCRAM', 'OAUTHBEARER']})
+        elif feature == Features.schema_id_validation:
+            self.redpanda.set_cluster_config(
+                {'enable_schema_id_validation': 'compat'})
+        elif feature == Features.rbac:
+
+            # NOTE(oren): make sure the role has propagated to every node since we don't know
+            # where the get_enterprise request will go
+            def has_role(r: str):
+                return all(
+                    len(
+                        RolesList.from_response(
+                            self.admin.list_roles(filter=r, node=n)).roles) > 0
+                    for n in self.redpanda.nodes)
+
+            self.admin.create_role('dummy')
+            wait_until(lambda: has_role('dummy'),
+                       timeout_sec=30,
+                       backoff_sec=1)
+        elif feature == Features.fips:
+            self.redpanda.rolling_restart_nodes(
+                self.redpanda.nodes,
+                override_cfg_params={
+                    'fips_mode':
+                    'permissive',
+                    "openssl_config_file":
+                    self.redpanda.get_openssl_config_file_path(),
+                    "openssl_module_directory":
+                    self.redpanda.get_openssl_modules_directory()
+                })
+        else:
+            assert False, f"Unexpected feature={feature}"
+
+        expect_status = EnterpriseLicenseStatus.valid if with_license else EnterpriseLicenseStatus.not_present
+        rsp = self.admin.get_enterprise_features().json()
+        enabled = {f['name']: f['enabled'] for f in rsp['features']}
+        for f in Features:
+            if f == feature:
+                assert enabled[
+                    f.
+                    name], f"expected {f} enabled, got {json.dumps(enabled, indent=1)}"
```

Comment:
nit: yapf... I'd split the line to make this more readable
```suggestion
                assert enabled[f.name], \
                    f"expected {f} enabled, got {json.dumps(enabled, indent=1)}"
```

Reply:
Nice, I didn't realize that Python would take that kind of thing. very `bash`y 🤷